src/librustdoc/html/templates/page.html

```diff
@@ -66,6 +66,7 @@
     {{- layout.external_html.in_header | safe -}}
 </head> {#- -#}
 <body class="rustdoc {{page.css_class}}"> {#- -#}
+    <a id="top"></a>
```

Comment:
Since https://github.com/rust-lang/rust/pull/92043 hasn't been merged yet, please add this ID in the ID map. ;)